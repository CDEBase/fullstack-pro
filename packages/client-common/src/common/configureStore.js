// @flow
import configureMiddleware from './configureMiddleware';
import configureReducer from './configureReducer';
import configureStorage from './configureStorage';
import { applyMiddleware, createStore, compose } from 'redux';
import { persistStore, autoRehydrate } from 'redux-persist';

/*
  Removed .defualt from require('..').default
  Still using routerMiddleware but it was removed in source
  Add enhancer
 */

type Options = {
  initialState: Object,
  platformDeps?: Object,
  platformMiddleware?: Array<Function>,
};

const configureStore = (options: Options) => {
  const {
    initialState,
    platformDeps = {},
    platformMiddleware = [],
  } = options;

  const reducer = configureReducer(initialState);

  const middleware = configureMiddleware(
    initialState,
    platformDeps,
    platformMiddleware,
  );

  // ======================================================
  // Store Enhancers
  // ======================================================
  const enhancers = [];
  if ((process.env.NODE_ENV !== 'production' || (Meteor.settings.public.logLevel === 'debug' && Meteor.isClient)) && window.devToolsExtension) {
    const devToolsExtension = window.devToolsExtension;
    if (typeof devToolsExtension === 'function') {
      enhancers.push(devToolsExtension());
    }
  }


  const store = createStore(
    reducer,
    initialState,
    compose(
      applyMiddleware(...middleware),
      autoRehydrate(),
      ...enhancers,
    ),
  );

  if (platformDeps.storageEngine) {
    const config = configureStorage(
      initialState.config.appName,
      platformDeps.storageEngine,
    );
    persistStore(store, config);
  }

  // to inject reducers in future
  store.asyncReducers = {};


  // Enable hot reloading for reducers.
  if (module.hot && typeof module.hot.accept === 'function') {
    if (initialState.device.isReactNative) {
      // React Native for some reason needs accept without the explicit path.
      // facebook.github.io/react-native/blog/2016/03/24/introducing-hot-reloading.html
      module.hot.accept(() => {
        const configureReducer = require('./configureReducer').default;

        store.replaceReducer(configureReducer(initialState));
      });
    } else {
      // Webpack for some reason needs accept with the explicit path.
      module.hot.accept('./configureReducer', () => {
        const configureReducer = require('./configureReducer').default;

        store.replaceReducer(configureReducer(initialState));
      });
    }
  }

  return store;
};

export default configureStore;
