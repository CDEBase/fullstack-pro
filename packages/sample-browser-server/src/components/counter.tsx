import * as React from 'react'
import * as redux from 'redux'
import { connect } from 'react-redux'

import {
  incrementCounter,
  loadCount,
  saveCount,
  Action
} from '../actions'

import { Store } from '../reducers'

export type OwnProps = {
  label: string,
  store?: redux.Store<Store.All>
}

export type ConnectedState = {
  counter: { value: number }
  isSaving: boolean,
  isLoading: boolean,
  error: string,
}

export type ConnectedDispatch = {
  increment: (n: number) => void
  save: (n: number) => void
  load: () => void
}

const mapStateToProps = (state: Store.All, ownProps: OwnProps): ConnectedState => ({
  counter: state.counter,
  isSaving: state.isSaving,
  isLoading: state.isLoading,
  error: state.error,
})

// const mapDispatchToProps = (dispatch: redux.Dispatch<Store.All>): ConnectedDispatch => ({
//   increment: (n: number) =>
//     dispatch(incrementCounter(n)),
//   load: () =>
//     loadCount(null)(dispatch),
//   save: (value: number) =>
//     saveCount({ value })(dispatch),
// })

const mapDispatchToProps = (dispatch: redux.Dispatch<Store.All>): ConnectedDispatch => ({
  increment: (n: number) =>
    dispatch(incrementCounter(n)),
  load: () =>
    dispatch(loadCount(null)),
  save: (value: number) =>
    dispatch(saveCount({ value })),
})

class CounterComponent extends React.Component<ConnectedState & ConnectedDispatch & OwnProps, {}> {

  _onClickIncrement = (e: React.SyntheticEvent<HTMLButtonElement>) => {
    e.preventDefault()
    this.props.increment(1)
  }

  _onClickSave = (e: React.SyntheticEvent<HTMLButtonElement>) => {
    e.preventDefault()
    if (!this.props.isSaving) {
      this.props.save(this.props.counter.value)
    }
  }

  _onClickLoad = (e: React.SyntheticEvent<HTMLButtonElement>) => {
    e.preventDefault()
    if (!this.props.isLoading) {
      this.props.load()
    }
  }

  render () {
    const { counter, label, isSaving, isLoading, error } = this.props
    return <form>
      <legend>{label}</legend>
      <pre>{JSON.stringify({ counter, isSaving, isLoading }, null, 2)}</pre>
      <button ref='increment' onClick={this._onClickIncrement}>click me!</button>
      <button ref='save' disabled={isSaving} onClick={this._onClickSave}>{isSaving ? 'saving...' : 'save'}</button>
      <button ref='load' disabled={isLoading} onClick={this._onClickLoad}>{ isLoading ? 'loading...' : 'load'}</button>
      { error ? <div className='error'>{error}</div> : null }
    </form>
  }
}

// https://github.com/DefinitelyTyped/DefinitelyTyped/issues/8787
export const Counter: React.ComponentClass<OwnProps> = connect(mapStateToProps, mapDispatchToProps)(CounterComponent)
