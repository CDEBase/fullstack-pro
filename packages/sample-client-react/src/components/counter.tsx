import * as React from 'react';

export interface ICounterProps {
    label: string;
    counter: { value: number };
    isSaving: boolean;
    isLoading: boolean;
    error: string;
    increment: (n: number) => void;
    save: (n: number) => void;
    load: () => void;
}

export interface ICounterState {
}

export class CounterComponent extends React.Component<ICounterProps, ICounterState> {

    private _onClickIncrement = (e: React.SyntheticEvent<HTMLButtonElement>) => {
        e.preventDefault();
        this.props.increment(1);
    }

    private _onClickSave = (e: React.SyntheticEvent<HTMLButtonElement>) => {
        e.preventDefault();
        if (!this.props.isSaving) {
            this.props.save(this.props.counter.value);
        }
    }

    private _onClickLoad = (e: React.SyntheticEvent<HTMLButtonElement>) => {
        e.preventDefault();
        if (!this.props.isLoading) {
            this.props.load();
        }
    }

    public render(): JSX.Element {
        const { counter, label, isSaving, isLoading, error } = this.props;
        return (
            <form>
                <legend>{label}</legend>
                <pre>{JSON.stringify({ counter, isSaving, isLoading }, null, 2)}</pre>
                <button onClick={this._onClickIncrement}>click me!</button>
                <button disabled={isSaving} onClick={this._onClickSave}>{isSaving ? 'saving...' : 'save'}</button>
                <button disabled={isLoading} onClick={this._onClickLoad}>{isLoading ? 'loading...' : 'load'}</button>
                {error ? <div className="error">{error}</div> : null}
            </form>);
    }
}


