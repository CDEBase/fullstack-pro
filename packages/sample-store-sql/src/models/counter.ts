import * as Sequelize from 'sequelize';


export default (sequelize: Sequelize, DataTypes: Sequelize.DataTypes) => {
    const Count = sequelize.define('count', {
        name: DataTypes.STRING,
    });
};
