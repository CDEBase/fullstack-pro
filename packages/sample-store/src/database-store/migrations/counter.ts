import * as Knex from 'knex';

export const Counter_Table = 'count';
export const createCounter = async (driver) =>
    await driver.schema.createTable(Counter_Table, table => {
        table.increments();
        table.timestamps(false, true);
        table.integer('amount').notNull();
    });
export const dropCounter = async (driver) =>
    driver.schema.dropTable(Counter_Table);

export async function up(knex: Knex) {
    return createCounter(knex);
}

export async function down(knex: Knex) {
    return await dropCounter(knex);
}
