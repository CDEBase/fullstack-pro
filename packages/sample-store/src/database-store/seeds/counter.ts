import * as Knex from 'knex';

const initialAmount = 5;

export async function seed(knex: Knex) {
    await knex('count').truncate();

    return await knex('count').insert({ amount: initialAmount });
}
