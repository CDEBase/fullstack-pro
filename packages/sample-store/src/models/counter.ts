import { DataTypes,Sequelize } from 'sequelize';


export default (sequelize: Sequelize, dataTypes: typeof DataTypes) => {
    // const count = sequelize.define('count', {
    //     name: dataTypes.STRING,
    // });
};
