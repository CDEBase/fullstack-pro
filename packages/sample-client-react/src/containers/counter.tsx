import {
  incrementCounter,
  loadCount,
  saveCount,
  Action,
  Store,
} from '@sample/client-redux';
import { connect } from 'react-redux';
import { returntypeof } from 'react-redux-typescript';
import * as redux from 'redux';
import { CounterComponent, ICounterProps } from '../components';

export type CounterOwnProps = {
  label: string;
  store?: Store.Sample;
};

const mapStateToProps = (state: Store.Sample) => ({
  counter: state['@sample/counter'],
  isSaving: state['@sample/isSaving'],
  isLoading: state['@sample/isLoading'],
  error: state['@sample/error'],
});

const mapDispatchToProps = (dispatch: Function) => ({
  increment: (n: number) =>
    dispatch(incrementCounter(n)),
  load: () =>
    dispatch(loadCount(null)),
  save: (value: number) =>
    dispatch(saveCount({ value })),
});

const stateProps = returntypeof(mapStateToProps);
const dispatchProps = returntypeof(mapDispatchToProps);
type StateProps = typeof stateProps;
type DispatchProps = typeof dispatchProps;

// https://github.com/DefinitelyTyped/DefinitelyTyped/issues/8787
export const Counter: React.ComponentClass<CounterOwnProps> =
connect<StateProps, DispatchProps, CounterOwnProps>(mapStateToProps, mapDispatchToProps)(CounterComponent);

