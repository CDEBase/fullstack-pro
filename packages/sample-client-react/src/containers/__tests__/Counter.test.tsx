import 'jest';
import React from 'react'
import { shallow, mount } from 'enzyme';
import * as Sinon from 'sinon';
// tslint:disable-next-line no-unused-variable
import * as TestUtils from 'react-dom/test-utils'
import { createStore, combineReducers } from 'redux'
import configureStore from 'redux-mock-store'
import { Counter } from '../counter'
import { reducers, Store } from '@sample/client-redux'
import { Provider } from 'react-redux';
import './setup';


describe('components/Counter', () => {
  it('renders', () => {
    const store:any = createStore(combineReducers<Store.Sample>(reducers));

    expect(shallow(
      <Counter label='a counter!' store={store} />
    ).shallow()).toMatchSnapshot()
  })

  describe('COUNTER --- react-redux clicking "increment"', () => {
    let counter
    let store

    beforeEach(() => {
      store = createStore(combineReducers(reducers));
      counter = mount(
        <Provider store={store}>
          <Counter label='a counter!' />
        </Provider>
      );
    })


    it('+++ check Props after increments counter',  () => {
      const increment = counter.find('button').first()
      increment.simulate('click')
      increment.simulate('click')
      increment.simulate('click')


      const getText = () => counter.find('pre').text();
      expect(JSON.parse(getText()).counter.value).toBe(3)

    })
  })
})
