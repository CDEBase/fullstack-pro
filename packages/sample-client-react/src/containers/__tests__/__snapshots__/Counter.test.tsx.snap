// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Counter renders 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <form>
    <legend>
        a counter!
    </legend>
    <pre>
        {
          "counter": {
            "value": 0
          },
          "isSaving": false,
          "isLoading": false
        }
    </pre>
    <button
        onClick={[Function]}
    >
        click me!
    </button>
    <button
        disabled={false}
        onClick={[Function]}
    >
        save
    </button>
    <button
        disabled={false}
        onClick={[Function]}
    >
        load
    </button>
</form>,
  "nodes": Array [
    <form>
      <legend>
            a counter!
      </legend>
      <pre>
            {
              "counter": {
                "value": 0
              },
              "isSaving": false,
              "isLoading": false
            }
      </pre>
      <button
            onClick={[Function]}
      >
            click me!
      </button>
      <button
            disabled={false}
            onClick={[Function]}
      >
            save
      </button>
      <button
            disabled={false}
            onClick={[Function]}
      >
            load
      </button>
</form>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <CounterComponent
        counter={
                Object {
                        "value": 0,
                      }
        }
        error=""
        increment={[Function]}
        isLoading={false}
        isSaving={false}
        label="a counter!"
        load={[Function]}
        save={[Function]}
        store={
                Object {
                        "dispatch": [Function],
                        "getState": [Function],
                        "replaceReducer": [Function],
                        "subscribe": [Function],
                        Symbol(observable): [Function],
                      }
        }
        storeSubscription={
                Subscription {
                        "listeners": Object {
                          "notify": [Function],
                        },
                        "onStateChange": [Function],
                        "parentSub": undefined,
                        "store": Object {
                          "dispatch": [Function],
                          "getState": [Function],
                          "replaceReducer": [Function],
                          "subscribe": [Function],
                          Symbol(observable): [Function],
                        },
                        "unsubscribe": null,
                      }
        }
/>,
      "_debugID": 3,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": CounterComponent {
        "_onClickIncrement": [Function],
        "_onClickLoad": [Function],
        "_onClickSave": [Function],
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "counter": Object {
            "value": 0,
          },
          "error": "",
          "increment": [Function],
          "isLoading": false,
          "isSaving": false,
          "label": "a counter!",
          "load": [Function],
          "save": [Function],
          "store": Object {
            "dispatch": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
            Symbol(observable): [Function],
          },
          "storeSubscription": Subscription {
            "listeners": Object {
              "notify": [Function],
            },
            "onStateChange": [Function],
            "parentSub": undefined,
            "store": Object {
              "dispatch": [Function],
              "getState": [Function],
              "replaceReducer": [Function],
              "subscribe": [Function],
              Symbol(observable): [Function],
            },
            "unsubscribe": null,
          },
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 2,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <form>
          <legend>
                    a counter!
          </legend>
          <pre>
                    {
                      "counter": {
                        "value": 0
                      },
                      "isSaving": false,
                      "isLoading": false
                    }
          </pre>
          <button
                    onClick={[Function]}
          >
                    click me!
          </button>
          <button
                    disabled={false}
                    onClick={[Function]}
          >
                    save
          </button>
          <button
                    disabled={false}
                    onClick={[Function]}
          >
                    load
          </button>
</form>,
        "_debugID": 4,
        "_renderedOutput": <form>
          <legend>
                    a counter!
          </legend>
          <pre>
                    {
                      "counter": {
                        "value": 0
                      },
                      "isSaving": false,
                      "isLoading": false
                    }
          </pre>
          <button
                    onClick={[Function]}
          >
                    click me!
          </button>
          <button
                    disabled={false}
                    onClick={[Function]}
          >
                    save
          </button>
          <button
                    disabled={false}
                    onClick={[Function]}
          >
                    load
          </button>
</form>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <CounterComponent
    counter={
        Object {
            "value": 0,
          }
    }
    error=""
    increment={[Function]}
    isLoading={false}
    isSaving={false}
    label="a counter!"
    load={[Function]}
    save={[Function]}
    store={
        Object {
            "dispatch": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
            Symbol(observable): [Function],
          }
    }
    storeSubscription={
        Subscription {
            "listeners": Object {
              "notify": [Function],
            },
            "onStateChange": [Function],
            "parentSub": undefined,
            "store": Object {
              "dispatch": [Function],
              "getState": [Function],
              "replaceReducer": [Function],
              "subscribe": [Function],
              Symbol(observable): [Function],
            },
            "unsubscribe": null,
          }
    }
/>,
}
`;
